### ISSUE TITLE
Add angle-restricted light sources to SenseMap

### ISSUE THREAD

**Issue by Chris3606** [Chris3606/GoRogue#6]
Angle-restricted shadowcasting is implemented already in FOV, and RIPPLE is possible.  Thus, adding angle-restricted light sources (directional lights) to SenseMap should be fairly simple.

**Comment by bloodyraoul** [comment]
I am waiting this feature too.
In version 1.6, you have added the angle only for FOV, not for LightSources, right ?
you plan to add it ?
Thanks

**Comment by Chris3606** [comment]
Sorry, I thought I responded to this awhile ago.  1.6 has angle for FOV only, yes.  Since one of the options for sense sources is shadow-casting, some of the work for doing it in sense-mapping is already done, I just need to get around to the rest.  I've got a few additions that I'm trying to work on for the next release, and definately if you're waiting on it, I'll try to make sure this one makes it as well.

**Comment by Chris3606** [comment]
This is my next target to work on for GoRogue 1.7.  I need to refactor the way FOV and SenseSource are handled anyway, to avoid duplicating the entirety of the shadow-casting algorithm, and that makes a convenient time to port the existing FOV code for shadow-casting to SenseMap, so might as well knock it out...
